src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2334,9 +2349,69 @@ ManageWorkerPool(WorkerPool *workerPool)
 	if (failedConnectionCount >= 1)
 	{
 		/* do not attempt to open more connections after one failed */
-		return;
+		return false;
 	}
 
+	if (IsTooEarlyToOpenConnections(workerPool))
```

Comment:
this function name is confusing me. This is effectively part of the slow start logic, which should be called out more explicitly.

Maybe rename to WaitForSlowStart and move out of CanOpenNewConnection and into ManageWorkerPool?